evap/results/exporters.py

```diff
@@ -262,12 +291,22 @@ def write_questionnaire(self, questionnaire, evaluations_with_results, contribut
                 approval_count = 0
 
                 for grade_result in results[questionnaire.id]:
-                    if grade_result.question.id != question.id or not grade_result.has_answers:
+                    if grade_result.question.id != question.id or (
+                        isinstance(grade_result, RatingResult) and not grade_result.has_answers
+                    ):
                         continue
-                    values.append(grade_result.average * grade_result.count_sum)
-                    count_sum += grade_result.count_sum
-                    if grade_result.question.is_yes_no_question:
-                        approval_count += grade_result.approval_count
+
+                    # `filter_text_and_heading_questions` filters all text questions
+                    # `grade_result.question.id != question.id` filters the remaining HeadingResults because `question.is_heading_question` filters all heading questions
+                    # so here all `grade_result` from the OrderedDict values are only RatingResults with the same question as the current question which is not a heading question or a text question so it must be a rating question
+                    rating_result = cast(RatingResult, grade_result)
+                    assert rating_result.count_sum
+                    assert rating_result.average
+                    values.append(rating_result.average * rating_result.count_sum)
+                    count_sum += rating_result.count_sum
+                    if rating_result.question.is_yes_no_question:
+                        assert rating_result.approval_count
```

Comment:
I don't think these asserts do what you intended them to do. I think you wanted to do this:
```suggestion
                    assert hasattr(rating_result, "count_sum")
                    assert hasattr(rating_result, "average")
                    values.append(rating_result.average * rating_result.count_sum)
                    count_sum += rating_result.count_sum
                    if rating_result.question.is_yes_no_question:
                        assert hasattr(rating_result, "approval_count")
```